File: electronic_backend/src/main/java/com/project/electronic/Model/Category.java
package com.project.electronic.Model;

import jakarta.persistence.*;

import java.util.List;

@Entity
@Table(name = "category")
public class Category {

    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer Id;
    @Column(name = "Name")
    private String Name;

    @OneToMany(mappedBy = "category")
    private List<Product> products;

    public Integer getId() {
        return Id;
    }

    public String getName() {
        return Name;
    }

    public void setId(Integer id) {
        Id = id;
    }

    public void setName(String name) {
        Name = name;
    }
}

File: Electronic_Frontend/app/src/main/java/com/example/electronic_frontend/model/Category.kt
package com.example.electronic_frontend.model

class Category {

    private var categoryid: Int? = null


    private var categoryname: String? = null


    private val products: List<Product>? = null

    fun getId(): Int? {
        return categoryid
    }

    fun getName(): String? {
        return categoryname
    }

    fun setId(id: Int?) {
        categoryid = id
    }

    fun setName(name: String?) {
        categoryname = name
    }
}